org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/BuilderStateDiscarder.java

```diff
@@ -68,7 +68,9 @@ protected void touchProject(IProject project) {
 		} else {
 			Job touchJob = Job.create("Touch project " + project.getName(), progressMonitor -> {
 				try {
-					project.touch(progressMonitor);
+					if (project.isAccessible()) {
+						project.touch(progressMonitor);
```

Comment:
Looking at the call hierarchy, no one is making sure that the projects that are passed in are accessible. Thus it's probably a good idea to also wrap either the entirety of the function in a `project.isAccessible()` or to wrap line 64 with `project.isAccessible()`